drivers/spi/spi_sam.c

```diff
@@ -180,7 +180,7 @@ static void spi_sam_finish(Spi *regs)
 /* Fast path that transmits a buf */
 static void spi_sam_fast_tx(Spi *regs, const struct spi_buf *tx_buf)
 {
-	struct k_spinlock lock;
+	struct k_spinlock lock = {};
```

Comment:
Empty initialiser is a GNU extension and should not be used.

That aside, this lock should not be in the stack as @fabiobaltieri already pointed out above.

Reply:
Uh, this pattern is used pervasively in the tree; I count hundreds and hundreds of usages.  If you want to migrate us in that direction I think that's reasonable, but... let's not start NAKing PRs that are just writing Zephyr code in Zephyr style.

Reply:
Empty initialiser is not a thing according to the ISO C standard (at least pre-C23) and, as expected, not supported by some compilers (notoriously, IAR whose support is currently WIP).

In fact, there is a PR "fixing" all the empty initialiser usages because of that: https://github.com/zephyrproject-rtos/zephyr/pull/54379.

This is not a coding style issue; this is about writing valid and portable C code.

Reply:
Simple enough, I'll update to `{0}`